### DISCUSSION TITLE
reporting an error in working with rstudio quarto

### DISCUSSION THREAD

**Discussion by MohammadTaghizadeh** [quarto-dev/quarto-cli#6088] [Q&A]
### Description

I got an error in working with Rstudio Quarto:
For a simple example file when I click render:
ERROR: Error executing 'Invalid macro definition.': program not found

Unable to locate an installed version of R.
Install R from https://cloud.r-project.org/

what can I do?

**Comment by mcanouil** [comment]
First we are going to need more information, such as operating system, versions of the tools, etc., and the example you tried.
As is we can't help you.

So, could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by MohammadTaghizadeh** [reply]
I just use the newest version of Rstuio and R and for any example of qmd files i.e. hello.qmd I got this error. 
![image](https://github.com/quarto-dev/quarto-cli/assets/66325392/ce02f008-ae59-4180-84e4-447ce5f0dd30)

**Reply by mcanouil** [reply]
Still, what operating system?
Do you use custom `.Rprofile`?
What if you open a terminal (PowerShell, Bash, zsh, or whatever is in your system) and type `R`, does R open?

For short, the issue is likely about your setup rather than with Quarto.

**Reply by MohammadTaghizadeh** [reply]
my os is Windows 10 Pro and running R on cmd gave me: 
![image](https://github.com/quarto-dev/quarto-cli/assets/66325392/5f4d41bd-9691-4dab-8a04-2c02562594ed)

it gave an invalid micro definition but it worked.

**Reply by mcanouil** [reply]
Thanks for the additional information.

You seem to load a saved workspace, this is considered in many case as a bad practice because it can hide issues from past code.
In your project, remove the `.rdata` file.

Does R open if you open it in the Terminal within RStudio?

What's the "macro invalid definition" you have? I believe this to be the main issue.
This does not seem to come from R.

**Reply by MohammadTaghizadeh** [reply]
hello again,
I removed rdata file from the project still it didn't work, and for executing R from Rstudio terminal, I did it and the result was just like when I did it from an independent cmd (like the previous image).  

![image](https://github.com/quarto-dev/quarto-cli/assets/66325392/5f56fd22-7e98-4f2f-92bf-b8d2b8858916)

![image](https://github.com/quarto-dev/quarto-cli/assets/66325392/064e7478-d178-4257-ba7b-d63bb1a24741)

**Reply by mcanouil** [reply]
@cderv Do you have any ideas/insights about the "invalid macro definition" on Windows and the likely fact that this is the issue of Quarto erroring out earlier?

**Reply by cderv** [reply]
Nope I don't know really. Usually this type of error ("invalid macro definition") are within a C / C++ context to me. 

Though what I can see here is that the error is thrown right after running R at command line. So it seems to me RStudio IDE may be checking that R loads correctly, but see this two lines when starting R and it does not work. 

@MohammadTaghizadeh does the issue only happen when clicking Render button inside RStudio IDE ? 

Does `quarto render` works in terminal (inside or outside RStudio) ? 

It seems to me something related to your environment, or installation of R.
